src/viam/sdk/components/camera/camera.hpp

```diff
@@ -83,6 +84,66 @@ class Camera : public Component {
         response_metadata metadata;
     };
 
+    /// @brief UTF-8 encoding of 'DEPTHMAP' used in the header of FORMAT_RAW_DEPTH bytes payload.
+    static const uint64_t MAGIC_NUMBER = 0x44455054484D4150ULL;
+
+    /// @brief Number of bytes of the header for FORMAT_RAW_DEPTH payloads
+    static const auto HEADER_SIZE = 8 /*magic num*/ + 8 /*width*/ + 8 /*height*/;
+
+    /// @struct depth_map
+    /// @brief Represents the dimensions and depth values of a depth map.
+    ///
+    /// depth_map holds the width and height of a depth map, along with a vector
+    /// of actual depth values. Each depth value is a 16-bit unsigned integer representing
+    /// the distance from the camera to a point in the scene.
+    struct depth_map {
+        uint64_t width;                      ///< Width of the depth map in pixels.
+        uint64_t height;                     ///< Height of the depth map in pixels.
+        std::vector<uint16_t> depth_values;  ///< A flat vector of depth values.
```

Comment:
The `depth_map` type is the vocabulary type for the SDK. So I think it is worth ensuring that the interface that it presents offers good UX. As written, I still need to do index computation in order to obtain the depth value at a given `(x, y)` by doing `dm.depth_values[dm.height * x + y]` or something like that.

The C++ SDK already has a dependency on the xtensor library via the mlmodel service API. You could trivially represent this 2d array of uint16 with xtensor, and then you would get indexed access for free.

If xtensor feels like overkill for a 2d array there are probably other ways we can get a good 2d array concept rather than pushing the need to index into a linear array to the user.

Reply:
I may be wrong but I think we should keep it as a flat vector. ML frameworks like pytorch and tensorflow has utilities to convert vectors to the native data structs e.g. `torch.from_blob` or `tf.convert_to_tensor`. And then you can use something like `.reshape` with the width and height. I looked it up and it's actually quite a bit harder to make this conversion with xtensor. 

I feel like direct indexing is not a common use case with depth maps? Again, not a CV guy could be wrong lol